modin/core/dataframe/pandas/dataframe/protocol/column.py

```
@@ -0,0 +1,590 @@
+                self._col._partitions,
+                lambda df: df,
+                keep_partitioning=False,
+                lengths=new_row_lengths,
+            )
+            new_df = self._col.__constructor__(
+                new_partitions,
+                self._col.index,
+                self._col.columns,
+                new_row_lengths,
+                self._col._column_widths,
+            )
+            for length in new_df._row_lengths:
+                yield Column(
+                    PandasDataframe(
+                        self._col.mask(row_positions=range(length), col_positions=None),
+                        allow_copy=self._allow_copy,
+                        offset=offset,
+                    )
+                )
+                offset += length
+
+    def get_buffers(self) -> Dict[str, Any]:
+        """
+        Return a dictionary containing the underlying buffers.
+
+        Returns
+        -------
+        dict
+            - "data": a two-element tuple whose first element is a buffer
+              containing the data and whose second element is the data buffer's associated dtype.
+            - "validity": a two-element tuple whose first element is a buffer
+              containing mask values indicating missing data and
+              whose second element is the mask value buffer's
+              associated dtype. None if the null representation is not a bit or byte mask.
+            - "offsets": a two-element tuple whose first element is a buffer
+              containing the offset values for variable-size binary data
+              (e.g., variable-length strings) and whose second element is the offsets
+              buffer's associated dtype. None if the data buffer does not have
+              an associated offsets buffer.
+        """
+        buffers = {}
+        buffers["data"] = self._get_data_buffer()
+        try:
+            buffers["validity"] = self._get_validity_buffer()
+        except Exception:
```

let's use a specific exception type for this purpose

---

That is not a big deal because we catch an exception and continue a pipeline.

---

I believe we want to raise an exception if something went wrong during `self._get_smt_buffer()` rather than just returning a `None` which will cause an error in the import function (much away from where the error actually occurred).

If we will catch a specific exception type here we will avoid silencing arbitrary exceptions

---

Makes sense, fixed.